### PR TITLE
Introducing IsInf

### PR DESCRIPTION
**Pull request by bhushan23**
torch.isinf - checks element wise +/- inf implements #9132

### CODE REVIEW

**Comment by bhushan23** [comment]
@SsnL @vishwakftw please review

**Comment by vishwakftw** [comment]
Please add tests in `test_torch.py`

**Comment by bhushan23** [comment]
test_isinf added

**Comment by SsnL** [comment]
`test_isinf` doesn't seem added

**Comment by bhushan23** [comment]
Oh! It was missed! added !! 
We only need to create test method in TestTorch?

**Comment by vishwakftw** [comment]
Add the function to `docs/source/torch.rst`, just like `torch.isnan`.

**Comment by bhushan23** [comment]
Thanks @vishwakftw done

**Comment by bhushan23** [comment]
@SsnL @zou3519 can you approve the change?

File: torch/functional.py [hunk 1]
```diff
@@ -167,6 +168,28 @@ def isnan(tensor):
...
+    Arguments:
+        tensor (Tensor): A tensor to check
+
+    Returns:
+        Tensor: A ``torch.ByteTensor`` containing a 1 at each location of `+/-INF` elements.
+
+    Example::
+
+        >>> torch.isinf(torch.Tensor([1, float('inf'), 2]))
+         0
```
Comment (SsnL) [inline torch/functional.py]: This is old style print. Can you update to the new one?

Comment (bhushan23) [reply torch/functional.py]: updated

**Comment by bhushan23** [comment]
@SsnL printing with new format

**Comment by bhushan23** [comment]
@SsnL

File: test/test_torch.py [hunk 2]
```diff
@@ -4533,6 +4533,10 @@ def test_isnan(self):
         x = torch.Tensor([1, float('nan'), 2])
         self.assertEqual(torch.isnan(x), torch.ByteTensor([0, 1, 0]))
 
+    def test_isinf(self):
+        x = torch.Tensor([1, float('inf'), 2, float('-inf')])
```
Comment (zou3519) [inline test/test_torch.py]: nit: Could you add a test case for float('NaN')?

File: torch/functional.py [hunk 3]
```diff
@@ -167,6 +168,25 @@ def isnan(tensor):
     return tensor != tensor
 
 
+def isinf(tensor):
+    r"""Returns a new tensor with boolean elements representing if each element is `+/-INF` or not.
```
Comment (zou3519) [inline torch/functional.py]: Nit: the elements aren't boolean, they're either 0 or 1.

File: torch/functional.py [hunk 4]
```diff
@@ -167,6 +168,25 @@ def isnan(tensor):
...
+
+    Arguments:
+        tensor (Tensor): A tensor to check
+
+    Returns:
+        Tensor: A ``torch.ByteTensor`` containing a 1 at each location of `+/-INF` elements.
```
Comment (zou3519) [inline torch/functional.py]: nit: And 0 otherwise

File: torch/functional.py [hunk 5]
```diff
@@ -167,6 +168,25 @@ def isnan(tensor):
...
+
+    Example::
+
+        >>> torch.isinf(torch.Tensor([1, float('inf'), 2, float('-inf')]))
+        tensor([ 0,  1,  0,  1], dtype=torch.uint8)
+    """
+    if not torch.is_tensor(tensor):
+        raise ValueError("The argument is not a tensor")
```
Comment (zou3519) [inline torch/functional.py]: nit: It would be nice to include the object in the error message. Something like `"Not a tensor: " + str(tensor)`

**Review by zou3519 — COMMENTED** [review]
The code for isinf is a one-liner, so I'm not sure it is worth adding it to the API.

Other than that, I left a few nits below: the code looks good.

**Comment by SsnL** [comment]
We have `isnan` so `isinf` seems fine. Also np has it IIRC. We should move them to ATen one day though.

**Comment by vadimkantorov** [comment]
`isfinite` should also be a one-liner (at least an inefficient one) - it's also supported by numpy/tf, but with `isinf` the pain is already less anyway :)

File: torch/functional.py (hunk 3 above)
Comment (bhushan23) [reply torch/functional.py]: I think we should keep it for consistency, anyway it is indeed boolean element right?

File: torch/functional.py (hunk 4 above)
Comment (bhushan23) [reply torch/functional.py]: Added

File: torch/functional.py (hunk 5 above)
Comment (bhushan23) [reply torch/functional.py]: Added

**Comment by bhushan23** [comment]
@zou3519 
1. Dumping input in case not a tensor
2. Updated comment
3. Keeping ByteTensor sentence in comment for consistency with other methods and also because it is indeed ByteTensor. We are stating in next line that input is either 1/0

**Comment by vishwakftw** [comment]
This has already been fixed in a previous PR, I believe.

File: torch/functional.py [hunk 6]
```diff
@@ -167,6 +168,25 @@ def isnan(tensor):
...
+    Arguments:
+        tensor (Tensor): A tensor to check
+
+    Returns:
+        Tensor: A ``torch.ByteTensor`` containing a 1 at each location of `+/-INF` elements and 0 otherwise
+
+    Example::
+
+        >>> torch.isinf(torch.Tensor([1, float('inf'), 2, float('-inf'), float('nan')]))
+        tensor([ 0,  1,  0,  1, 0], dtype=torch.uint8)
```
Comment (vishwakftw) [inline torch/functional.py]: nit: spacing error

File: docs/source/torch.rst [hunk 7]
```diff
@@ -223,6 +223,7 @@ Comparison Ops
 .. autofunction:: ge
 .. autofunction:: gt
 .. autofunction:: isnan
+.. autofunction:: isinf
```
Comment (vishwakftw) [inline docs/source/torch.rst]: Please place this above `isnan` to maintain the order.

File: torch/functional.py [hunk 8]
```diff
@@ -9,6 +9,7 @@
     'btrifact',
     'btriunpack',
     'isnan',
+    'isinf',
```
Comment (vishwakftw) [inline torch/functional.py]: Same here.

**Comment by bhushan23** [comment]
are we done here?

**Comment by zou3519** [comment]
@bhushan23 thank you for the reminder. Let me give this one more look through.

**Review by zou3519 — APPROVED** [review]
LGTM!

**Comment by bhushan23** [comment]
@apaszke @colesbury @ezyang @gchanan @soumith need one of yours approval too

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**